crates/module-system/module-implementations/sov-evm/src/evm/sov_evm.rs

```diff
@@ -0,0 +1,102 @@
+use delegate::delegate;
+use revm::{
+    context::{ContextError, ContextSetters, ContextTr, Evm, FrameStack},
+    handler::{
+        evm::FrameTr, instructions::EthInstructions, EthFrame, EthPrecompiles, EvmTr,
+        FrameInitOrResult, ItemOrResult,
+    },
+    inspector::{InspectorEvmTr, JournalExt},
+    interpreter::interpreter::EthInterpreter,
+    Database, Inspector,
+};
+
+/// TODO
+#[derive(Debug)]
+pub struct SovEvm<CTX, INSP>(
+    pub  Evm<
+        CTX,
+        INSP,
+        EthInstructions<EthInterpreter, CTX>,
+        EthPrecompiles,
+        EthFrame<EthInterpreter>,
+    >,
+);
+
+impl<CTX: ContextTr, INSP> SovEvm<CTX, INSP> {
+    /// TODO
+    pub fn new(ctx: CTX, inspector: INSP) -> Self {
+        Self(Evm {
+            ctx,
+            inspector,
+            instruction: EthInstructions::new_mainnet(),
+            precompiles: EthPrecompiles::default(),
+            frame_stack: FrameStack::new(),
+        })
+    }
+}
+
+impl<CTX: ContextTr, INSP> EvmTr for SovEvm<CTX, INSP>
+where
+    CTX: ContextTr,
+{
+    type Context = CTX;
+    type Instructions = EthInstructions<EthInterpreter, CTX>;
+    type Precompiles = EthPrecompiles;
+    type Frame = EthFrame<EthInterpreter>;
+
+    delegate! {
```

Comment:
We generally avoid macros that alter Rust’s standard syntax (we use it ocassionally but only when neccsesary). I know this feels repetitive, but let’s stick with providing a manual implementation instead?